### DISCUSSION TITLE
Is There a Way to Force Re-indexing Backlinks?

### DISCUSSION THREAD

**Discussion by zicklag** [SebastianZimmer/neno#65] [Q&A] [answered]
Hey there! I've been using Neno for taking notes for the last while and just recently in the last few days I noticed that it's not updating the backlinks for some notes all of the sudden.

I'm not exactly sure how to reproduce, because I have 220 notes, and it only just started happening, and I can't reproduce with a quick test in a new folder.

Do you have any guidance on what I could do to get the backlinks to update, or what I can do to troubleshoot?

**Edit:** I just realized that the backlinks will work if I use the `[[my-page]]` format, but not if I use the `/my-page`.

I'm using the `/my-page` format for everything to be compatible with the subtext spec.

**Comment by polyrainbow** [comment]
Hey there,
I'm happy trying to help finding the root cause of this issue.

I quickly did a test and also could not reproduce it, neither by using slaslinks or wikilinks.

You can re-index backlinks by just reloading the page. All indexes are re-created on each page load from the subtext files on disk. There is no browser storage cache or whatsoever.

Do the backlinks re-appear after startup? Or are they still missing? If they re-appear, this must be an issue with the backlink index updater during runtime. If not, it's a more general issue with how backlink indexing works.

**Reply by zicklag** [reply]
Thanks! Nothing changes when I reload the page, even with a ctrl+shift+R for a forced reload.

But, I was just able to reproduce with two _really_ tiny notes, so hopefully that helps!

[nenotest.zip](https://github.com/user-attachments/files/17193531/nenotest.zip)

**Reply by polyrainbow** [reply]
Yes, I can reproduce it. Indeed a bug. I'll investigate...
Thank you!

**Comment by polyrainbow** [comment, marked as answer]
The fix is included in https://github.com/polyrainbow/neno/pull/66 and in [v8.6.3](https://github.com/polyrainbow/neno/releases/tag/v8.6.3). When you refresh the app a few times and give the service worker some time to update all assets, you should get it.

Two unrelated remarks:

> I'm using the /my-page format for everything to be compatible with the subtext spec.

Since there is no active development on the [original Subtext spec](https://github.com/subconsciousnetwork/subtext) anymore, I have [forked](https://github.com/polyrainbow/subtext/) it and I am planning to make Wikilinks part of the spec. The original spec authors already started an [exploration](https://github.com/polyrainbow/subtext/blob/main/archive/explorations/wikilink.md) for this.

I'm following your work on the Leaf Protocol and weird.one. Great stuff!

**Reply by zicklag** [reply]
> The fix is included in https://github.com/polyrainbow/neno/pull/66 and in [v8.6.3](https://github.com/polyrainbow/neno/releases/tag/v8.6.3). When you refresh the app a few times and give the service worker some time to update all assets, you should get it.

Awesome it works, thanks!

> I'm following your work on the Leaf Protocol and weird.one. Great stuff!

Thanks! :grinning: 

> Since there is no active development on the [original Subtext spec](https://github.com/subconsciousnetwork/subtext) anymore, I have [forked](https://github.com/polyrainbow/subtext/) it and I am planning to make Wikilinks part of the spec.

Oh, cool! My biggest reason for complying exactly with subtext was just to make parsing easier if I needed to migrate to another note tool, most-likely if I wrote my own using eventually using Leaf. :smile: But I definitely think wiki-style links are a good idea for the Subtext spec. 👍️
